src/main.rs

```diff
@@ -1150,6 +1164,43 @@ mod tests {
                 3
             );
 
+            let ip = guest.network.l2_guest_ip.clone();
+            let ip_v: Vec<i32> = ip.split('.').map(|s| s.parse().unwrap()).collect();
+            let ipsrc = format!("{}.{}.{}.{}", ip_v[0], ip_v[1], ip_v[2], ip_v[3] + 1);
+
+            let mac = guest.network.l2_guest_mac.clone();
+            let mac_v: Vec<u8> = mac
+                .split(':')
+                .map(|s| u8::from_str_radix(&s, 16).unwrap())
+                .collect();
+
+            let dest_mac = MacAddr::new(mac_v[0], mac_v[1], mac_v[2], mac_v[3], mac_v[4], mac_v[5]);
+
+            // Generate a UDP packet with data
+            let mut pkt_buf = [0u8; 1500];
+            let pkt = packet_builder!(
+                 pkt_buf,
+                 ether({set_destination => dest_mac, set_source => MacAddr(10,1,1,1,1,2)}) /
+                 ipv4({set_destination=> ipv4addr!(ip), set_source=> ipv4addr!(ipsrc) }) /
+                 udp({set_source => 4444, set_destination => 4444}) /
+                 payload({"Hello World through vhost-user-net!\n".to_string().into_bytes()})
+            );
+            let socket = UdpSocket::bind("127.0.0.1:23456").expect("couldn't bind to address");
+            socket
+                .send_to(&pkt.packet(), "127.0.0.1:4444")
+                .expect("couldn't send data");
+
+            let nc_cmd = format!("nc -u -l {} 4444 > net.log &", ip);
+            guest.ssh_command(&nc_cmd)?;
+
+            thread::sleep(std::time::Duration::new(1, 0));
```

Comment:
It's possible that you see different result between your local machine and the CI because it takes more time for the CI to run the same commands. Keep in mind the CI is running nested, which slows everything down.

Reply:
1 second doesn't seem very long at all.